src/main/java/uk/gov/hmcts/reform/draftstore/info/UnencryptedDraftsInfoContributor.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.draftstore.info;
+
+import org.springframework.boot.actuate.info.Info;
+import org.springframework.boot.actuate.info.InfoContributor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.draftstore.data.DraftStoreDAO;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.time.LocalDateTime.now;
+
+@Component
+public class UnencryptedDraftsInfoContributor implements InfoContributor {
+
+    private final DraftStoreDAO repo;
+
+    private LocalDateTime lastCheckDate = null;
+    private List<Map<String, Object>> lastResult = new ArrayList<>();
+
+    public UnencryptedDraftsInfoContributor(DraftStoreDAO repo) {
+        this.repo = repo;
+    }
+
+    @Override
+    public void contribute(Info.Builder builder) {
+        if (lastResult.isEmpty() || shouldRefresh()) {
+            this.lastResult = repo.getUnencryptedDrafts();
+            this.lastCheckDate = now();
+        }
+
+        builder.withDetail("unencrypted_drafts", this.lastResult);
+    }
+
+    private boolean shouldRefresh() {
+        return Optional
+            .ofNullable(lastCheckDate)
+            .map(x -> x.plusMinutes(1).isBefore(now()))
+            .orElse(true);
```

Comment:
```
if (lastCheckDate != null) {
  return lastCheckDate.plusMinutes(1).isBefore(now());
} else {
  return true;
}
```
? 
Using `Optional` to avoid writing an `if` statement often makes the code look awkward and harder to read, like here in my opinion. Main purpose of `Optional` are "blank" return values, not avoiding `!= null` at all costs.

Reply:
Or `return lastCheckDate == null || lastCheckDate.plusMinutes(1).isBefore(now())`. Not too important as it reads well to me as it is.